pandas/core/indexing.py

```diff
@@ -2032,7 +2032,16 @@ def __getitem__(self, key):
                 raise ValueError("Invalid call for scalar access (getting)!")
 
         key = self._convert_key(key)
-        return self.obj._get_value(*key, takeable=self._takeable)
+        try:
+            return self.obj._get_value(*key, takeable=self._takeable)
+        except KeyError as err:
+            if isinstance(self.obj.index, ABCMultiIndex):
+                raise KeyError(
+                    f"Detected KeyError {err}, indexing with {key} "
+                    "failing for MultiIndex"
```

Comment:
doing this on master, the DataFrame case is giving a reasonable message, while the Series case is unhelpful.  Maybe condition this on `self.ndim == 1` and add a suggestion "pass a tuple"

Reply:
It the Series case goes down a different code path where an exception is raised in `pandas/utils/_validators.py`. Would adding an except for TypeError with a more informative error message work?

```
In [3]: ser.at["a_row", "A"]                                                                                  
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-2aec4c8078ba> in <module>
----> 1 ser.at["a_row", "A"]

/workspaces/pandas-aykhan/pandas/core/indexing.py in __getitem__(self, key)
   2090             return self.obj.loc[key]
   2091 
-> 2092         return super().__getitem__(key)
   2093 
   2094     def __setitem__(self, key, value):

/workspaces/pandas-aykhan/pandas/core/indexing.py in __getitem__(self, key)
   2035         try:
   2036             # import pdb; pdb.set_trace()
-> 2037             return self.obj._get_value(*key, takeable=self._takeable)
   2038         except KeyError as err:
   2039             if isinstance(self.obj.index, ABCMultiIndex):

TypeError: _get_value() got multiple values for argument 'takeable'
```